Automattic/jetpack
# Sync: Use new rest api endpoint for enabled sites

## Proposed changes:

Adds the ability to send Sync actions via the WPCOM REST API endpoint`v2/sites/site_id/jetpack-sync-actions`.

- It uses the `Settings:is_wpcom_rest_api_enabled` feature flag created in https://github.com/Automattic/jetpack/pull/36118 to determine whether to send the actions using REST API endpoint or the existing XML-RPC.
- Added logic to make sure we can sync an action through rest when enabling the Setting. If unable, the Setting gets disabled.

### Other information:


## Jetpack product discussion


p9dueE-7eh-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






Instructions in D141206-code should cover the basics.
Try also sending different types of errors from WPCOM to see how they are handled. This has been covered in Unit tests but manual testing is highly appreciated.
